test/MathOptInterface/MOI_Wrapper.jl

```diff
@@ -583,4 +583,125 @@ end
+        MIPPRESOLVE  = 0
+    )
+    # The variables: x[1:100], Bin
+    x, _ = MOI.add_constrained_variables(model, fill(MOI.ZeroOne(), 100))
+    # The objective function: maximize sum(profit' * x)
+    objf = MOI.ScalarAffineFunction(
+        MOI.ScalarAffineTerm.(profit, x), 0.0
+    )
+    MOI.set(
+        model,
+        MOI.ObjectiveFunction{MOI.ScalarAffineFunction{Float64}}(),
+        objf,
+    )
+    MOI.set(model, MOI.ObjectiveSense(), MOI.MAX_SENSE)
+    # The capacity constraint: sum(weight' * x) <= capacity
+    MOI.add_constraint(
+        model, MOI.ScalarAffineFunction(MOI.ScalarAffineTerm.(
+            weight, x), 0.0
+        ),
+        MOI.LessThan(capacity)
+    )
+
+    # FIRST RUN: get the optimal value. Check that the optimal
+    # value was not discovered at the root node.
+    MOI.optimize!(model)
+
+    solution = MOI.get(model, MOI.VariablePrimal(), x)
+    computed_obj_value = profit' * solution
+    obtained_obj_value = MOI.get(model, MOI.ObjectiveValue()) :: Float64
+    @test isapprox(
+        obtained_obj_value, computed_obj_value; rtol = rtol, atol = atol
+    )
+    @test isapprox(9945.0, computed_obj_value; rtol = rtol, atol = atol)
+    # Using MOI 9.22, so MOI.RawOptimizerAttribute was not available.
+    node_solution_was_found = Xpress.getintattrib(
+        model.inner, Xpress.Lib.XPRS_MIPSOLNODE
+    ) :: Cint
+    #=
+    MOI.get(
+        model, MOI.RawOptimizerAttribute("MIPSOLNODE")
+    ) :: CInt
+    =#
+    @test node_solution_was_found > 1
+
+    # SECOND RUN: run without MIP-start and only searching the first node.
+    # Should give a worse solution than the previous one.
+    MOI.set(model, MOI.RawParameter("MAXNODE"), 1)
+
+    MOI.optimize!(model)
+    @show model.callback_state
+
+    # One node may not be enough to even get any solution.
+    if MOI.get(model, MOI.PrimalStatus()) == MOI.FEASIBLE_POINT
+        solution2 = MOI.get(model, MOI.VariablePrimal(), x)
+        computed_obj_value2 = profit' * solution2
+        obtained_obj_value2 = MOI.get(model, MOI.ObjectiveValue()) :: Float64
+        @test isapprox(
+            obtained_obj_value2, computed_obj_value2; rtol = rtol, atol = atol
+        )
+        # There should be at least one unit of difference.
+        @test obtained_obj_value1 > obtained_obj_value2 + 0.5
+    end
+
+    # THIRD RUN: run with MIP-start and searching only the root node.
+    # Should find the optimal solution impossible to get in one node.
+    MOI.set(model, MOI.VariablePrimalStart(), x, solution)
+
+    # This postsolve is necessary because of an unrelated bug. Apparently,
+    # if Xpress is stopped because MAXNODE and has no feasible solution,
+    # then MOI.optimize! will not call postsolve over it, however calling
+    # postsolve is necessary otherwise a new call to MOI.optimize will
+    # trigger error 707 ("707 Error: Function cannot be called during the
+    # global search, except in callbacks.").
+    Xpress.postsolve(model.inner)

```

Comment:
This is weird indeed we should be checking if the problem is in presolve state at:
https://github.com/jump-dev/Xpress.jl/blob/8caf09af248514edb26728288ee39f9b44e6e076/src/MOI/MOI_wrapper.jl#L2362

Reply:
The flags PRESOLVE and MIPPRESOLVE are set to zero, so the problem should not be in presolved state anyway. I have run the code printing the `PRESOLVESTATE` bits just before my call to `Xpress.postsolve` which I commented, the output result is:

```
model.callback_state = Xpress.CB_NONE
is_lppres = false
is_mippre = false
is_solava = true
MIP Start testing: Error During Test at /home/henrique/AreaDeTrabalho/Xpress.jl/test/MathOptInterface/MOI_Wrapper.jl:590
  Got exception outside of a @test
  XpressError(32): Subroutine not completed successfully, possibly due to invalid argument. Unable to call `Xpress.addmipsol`:
  
  707 Error: Function cannot be called during the global search, except in callbacks.
```

So the problem seem to me like Xpress get stuck not in presolved state but in global search state (it thinks the optimization is yet in course) and somehow `postsolve` drags it outside of this state. What is interesting, because if I understood correctly, `postsolve` should error when the model is not in presolved state but here it gives no error and prevents the 707 error in the following `addmipsol` call.

Anyway, probably this should be tacked in another PR/issue. For these tests, can we keep them this way (calling `postsolve`), or should we try to change the tests so them do not step in this unrelated bug?

Reply:
Also, call me paranoid, but the Xpress documentation is suspiciously vague about `postsolve` uses, it says:

> A problem is left in a presolved state whenever a LP or MIP optimization does not complete.

One problem is the "a" in "a presolved state", there are many kinds of presolved states? Other is that the documentation seems to give the impression that even if the problem was not presolved to begin with (this is, no rows were changed or even presolve enabled) then stopping the optimization may just leave the solver in "a presolved state".

Reply:
Specially because this would be on par with how `mipoptimize` works:

> If the function returns without having completed the search for an optimal solution, the search can be resumed from where it stopped by calling problem.mipoptimize again.

Which is probably the source of this problem.

Reply:
More vague language that seems to indicate that the solver always end up in "presolved state" when the optimization is interrupted: https://www.fico.com/fico-xpress-optimization/docs/latest/mosel/mosel_lang/dhtml/postsolve.html

Reply:
yeah, I agree, docs are terrible.
I tried to always postsolve in `optimize!`, however, contrasting to what docs say it breaks sometimes if the model is already postsolved.

You can just leave it as is, and open an issue for us to track this in the future.

Reply:
Done: https://github.com/jump-dev/Xpress.jl/issues/150